src/components/RegistrationList.tsx

```diff
@@ -134,6 +161,25 @@ export const RegistrationListItemContent = ({
 
       {canEditRegistration && (
         <Box sx={{ display: 'flex', alignItems: 'start', gap: '0.5rem' }}>
+          {location.pathname.includes(UrlPathTemplate.MyPageMyResults) && (
+            <IconButton
+              title={t('my_page.my_profile.add_promoted_publication')}
+              data-testid={dataTestId.myPage.addPromotedPublicationButton}
+              disabled={isMutating}
+              onClick={() => {
+                if (isPromotedPublication) {
+                  mutatePromotedPublications.mutate(
+                    promotedPublications.filter((promotedPublicationId) => promotedPublicationId !== id)
+                  );
+                } else {
+                  mutatePromotedPublications.mutate([...(promotedPublications ?? []), id]);
```

Comment:
2 plasser vi har fallback/sjekk for tomt array for `promotedPublications` nå. Vil det forenkles om vi defaulter til tom array der vi dectronstructer props, eller har det noen dumme ringvirkinger? (`promotedPublications = []`)

Reply:
Funka greit det :+1: